### PR TITLE
Feature/drivers

### PR DESCRIPTION
**Pull request by fpdotmonkey**
Stuff that was meant to go into #52, but didn't for some reason

Run this to test:
```bash
./setup.sh
source ./venv/bin/activate
./runtests.sh
```

### CODE REVIEW

**Comment by Judoguy12** [comment]
@fpdotmonkey Has this been approved by @mmroden ?

**Comment by capsulecorplab** [comment]
> Stuff that was meant to go into #52, but didn't for some reason
> 
> Run this to test:
> 
> ```shell
> ./setup.sh
> source ./venv/bin/activate
> ./runtests.sh
> ```

Ran the above steps. The `runtests.sh` script doesn't seem to work inside a virtual environment. Also, `pip3 install -r requirements.txt` should be after `source ./venv/bin/activate`, as to install python dependencies within the virtual environment. Perhaps update the documentation to:
```shell
./setup.sh
source ./venv/bin/activate
pip3 install -r requirements.txt
python3 -m unittests -v
```

**Comment by mmroden** [comment]
No, the pip3 line goes in setup.sh.  The venv won't be active after running the script, so you have to activate it, and you may or may not want to run the tests.

**Comment by capsulecorplab** [comment]
If you run `pip3 install -r requirements.txt` before `source ./venv/bin/activate`, then you end up installing python dependencies to your global python interpreter's package directory, which is fine if you don't plan on running any other python application, but that would defeat the purpose of using a virtual environment.

**Review by capsulecorplab — APPROVED** [review]
Looked over and ran `setup.sh` and `runtests.sh` on my system. LGTM

File: 02_Development/Display/constants.py [hunk 1]
```diff
@@ -1,15 +1,25 @@
 # Constants for the display software for the
 # Tetra Biodistributed Ventillator Splitter
 
+PRESSURE_SENSOR_MUX_ADDRESS = 0x70
+FLOW_SENSOR_MUX_ADDRESS = 0x74
+
 NUMBER_OF_PATIENTS = 4
```
Comment (hydronics2) [inline 02_Development/Display/constants.py:7]: Is their a way we can gauge this from the sensor values we get back? Or are we going to just show data for four patients regardless? Maybe change it to max_number_of_patients as needed.

Comment (hydronics2) [reply 02_Development/Display/constants.py:7]: probably a bigger question than just a constant to determine the array size, eh?

File: Display/Documentation/HowToUseSPL06_007.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- For the mux select, I write a [0-4] integer to 0x70 (there's also a mux on 0x74, but I don't have anything connected to it right now, so it doesn't do anything.
+- I then reset the sensor to clear any previous state by writing 0b1001 to 0x0C
+- After that, I set the sensor into standby mode by writing 0 to 0x08
+- Then in the case of the tests I'm having issues with, I set the sensor to measure pressure at 1Hz with 16 measurements per sample by writing 0 | 0b0100 to 0x06.  Because there are more than 8 measurements per sample, I also set the sensor to bit shift the pressure by writing 0b0100 to 0x09.
+- I do the same to get the temperature, but now it's 1Hz 1 sample, so 0 | 0 gets written to 0x07.
+- Then I set the op mode to "command-pressure" by writing 0b0001 to 0x08 so I can read out what the current pressure is.
+- After waiting a few milliseconds for the measurement to be ready, I read addresses 0x00, 0x01, and 0x02 to get the pressure measurement.
+- Then the op mode is changed to "command-temperature" by writing 0b0010 to 0x08.
+- Finally, I read the temperature measurement from 0x03, 0x04, and 0x05
+
```
Comment (hydronics2) [inline Display/Documentation/HowToUseSPL06_007.md:16]: this is great. thanks for including this write-up! Great place to start from.

File: 02_Development/Display/features/gets_sensor_data.feature [hunk 3]
```diff
@@ -7,11 +7,12 @@ Feature: gets data from sensors
         Then the software will determine which tubes have both pressure and airflow sensing
         Then the software will return the state of the sensor to the user.
```
Comment (hydronics2) [inline 02_Development/Display/features/gets_sensor_data.feature:9]: we may consider adding the feature of indicating when the unit is unplugged. Their is an input on one of the channels of the ADC that measures input voltage. If the input voltage  measured by the ADC indicates 12.7V, we're plugged into the wall. If the input voltage measured by the ADC indicates 12V, we're on battery power.

**Review by hydronics2 — COMMENTED** [review]
Looks good!

There are some good instructions for setting up the PI, SSH into the device, and launch the software. I recommend someone trying this with the complete setup; that is, a PI, shield, and display.

see one-line comments as needed.

**Pull request merged**